sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCStreamWriteSuite.scala

```diff
@@ -0,0 +1,194 @@
+
+class JDBCStreamWriteSuite extends StreamTest with BeforeAndAfter {
+  import testImplicits._
+
+  val url = "jdbc:h2:mem:testdb"
+  val jdbcTableName = "stream_test_table"
+  val driverClassName = "org.h2.Driver"
+  val createTableSql = s"""
+      |CREATE TABLE ${jdbcTableName}(
+      | name VARCHAR(32),
+      | value LONG,
+      | PRIMARY KEY (name)
+      |)""".stripMargin
+
+  var conn: java.sql.Connection = null
+
+  val testH2Dialect = new JdbcDialect {
+    override def canHandle(url: String) : Boolean = url.startsWith("jdbc:h2")
+    override def isCascadingTruncateTable(): Option[Boolean] = Some(false)
+  }
+
+  before {
+    Utils.classForName(driverClassName)
+    conn = DriverManager.getConnection(url)
+    conn.prepareStatement(createTableSql).executeUpdate()
+  }
+
+  after {
+    conn.close()
+  }
+
+  test("Basic Write") {
+    withTempDir { checkpointDir => {
+        val input = MemoryStream[Int]
+        val query = input.toDF().map { row =>
+          val value = row.getInt(0)
+          TestData(s"name_$value", value.toLong)
+        }.writeStream
+          .format("jdbc")
+          .option(JDBCOptions.JDBC_URL, url)
+          .option(JDBCOptions.JDBC_TABLE_NAME, jdbcTableName)
+          .option(JDBCOptions.JDBC_DRIVER_CLASS, driverClassName)
+          .option("checkpointLocation", checkpointDir.getCanonicalPath)
+          .start()
+        try {
+          input.addData(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
+          query.processAllAvailable()
+        } finally {
+          query.stop()
+        }
+      }
+    }
+    val result = conn
+      .prepareStatement(s"select count(*) as count from $jdbcTableName")
+      .executeQuery()
+    assert(result.next())
+    assert(result.getInt("count") == 10)
+  }
+
+  test("Write sub columns") {
+    withTempDir { checkpointDir => {
+      val input = MemoryStream[Int]
+      val query = input.toDF().map { row =>
+        val value = row.getInt(0)
+        TestData(s"name_$value", value.toLong)
+      }.select("name").writeStream // write just one `name` column
+        .format("jdbc")
+        .option(JDBCOptions.JDBC_URL, url)
+        .option(JDBCOptions.JDBC_TABLE_NAME, jdbcTableName)
+        .option(JDBCOptions.JDBC_DRIVER_CLASS, driverClassName)
+        .option("checkpointLocation", checkpointDir.getCanonicalPath)
+        .start()
+      try {
+        input.addData(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
+        query.processAllAvailable()
+      } finally {
+        query.stop()
+      }
+    }
+    }
+    val result = conn
+      .prepareStatement(s"select count(*) as count from $jdbcTableName")
+      .executeQuery()
+    assert(result.next())
+    assert(result.getInt("count") == 10)
+  }
+
+  test("Write same data") {
+    withTempDir { checkpointDir => {
+      val input = MemoryStream[Int]
+      val query = input.toDF().map { row =>
+        val value = row.getInt(0)
+        TestData(s"name_$value", value.toLong)
```

Comment:
nit: can we `row.getLong(0)` directly?

Reply:
using row.getLong(0) directly will cause ClassCastException, part of the exception stack is as follows:
java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long
        at scala.runtime.BoxesRunTime.unboxToLong(BoxesRunTime.java:107)
        at org.apache.spark.sql.Row.getLong(Row.scala:232)
        at org.apache.spark.sql.Row.getLong$(Row.scala:232)
        at org.apache.spark.sql.catalyst.expressions.GenericRow.getLong(rows.scala:166)
Need to change the input data schema also, I will change the input data type, using "val  input = MemoryStream[Long]".

Reply:
Ah, then lets just do row.getInt.toLong. not a big deal though.